packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -5031,6 +5032,20 @@ class SemanticsConfiguration {
     _maxValueLength ??= child._maxValueLength;
     _currentValueLength ??= child._currentValueLength;
 
+    // Pick the highest heading level when merging.
+    //
+    // Heading levels are lower-is-higher (1 is the highest; 6 is the lowest),
+    // except 0, which means "not a heading", i.e. it's the lowest. Hence the
+    // complicated boolean expression.
+    final bool isChildHeadingLevelHigher =
```

Comment:
Why do we choose the highest from children? should we instead let the parent override children's heading like what we do for value or hint?

Reply:
My personal logic is that if a widget declares that it's H1, then it also probably uses a large text font it to communicate it. If you are merging multiple headings then the merged node will be no smaller than the largest node in the merge group. But I'm open to other interpretations of what "merging" means in this case. If there's an established convention, let's go with that.

Reply:
I think as of now all the other attributes are parent overriding the children's. I am more incline to preserve the behavior.

Also there is another mergeSemantics in getSemanticData, and it looks like the logic is wrong there as well.